monitoring/metrics.go

```diff
@@ -24,13 +24,13 @@ type metricsOptions struct {
 	labelMatchers      []*matchers.Matcher
 	waitMissingMetrics bool
 	skipMissingMetrics bool
-	waitBackoff        *backoff.Config
+	waitBackoff        *backoff.Backoff
 }
 
 // WithWaitBackoff is an option to configure a backoff when waiting on a metric value.
 func WithWaitBackoff(backoffConfig *backoff.Config) MetricsOption {
 	return func(o *metricsOptions) {
-		o.waitBackoff = backoffConfig
+		o.waitBackoff = backoff.New(context.Background(), *backoffConfig)
```

Comment:
Can we allow passing it? Maybe we can add another function `WithWaitBackoffCtx` to avoid too many changes

Reply:
Lets do this in a followup PR!